### PR TITLE
Update DG with Product Scope, NFRs and Glossary

### PR DESCRIPTION
**Pull request by ckcherry23**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -326,18 +326,48 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-3.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+**Technical requirements:**
+
+1.  The product should work on any _mainstream OS_ as long as it has Java `11` installed.
+
+**Quality requirements:**
+
+1.  A user with above average typing speed for plain English text should be able to accomplish most of the tasks using commands faster than with the mouse.
+
+**Constraints:**
```
Comment (NicsunXnus) [inline docs/DeveloperGuide.md:340]: I feel that constraints is vague. How about maintability requirement because testability is mentioned.

Comment (NicsunXnus) [reply docs/DeveloperGuide.md:340]: can reference here for others: https://www.guru99.com/non-functional-requirement-type-example.html

Comment (ckcherry23) [reply docs/DeveloperGuide.md:340]: I got `Constraints` from the concrete NFR examples given in the textbook: https://nus-cs2103-ay2122s2.github.io/website/se-book-adapted/chapters/requirements.html#non-functional-requirements

Another alternative we can have is `Testability`?

Lemme know your thoughts

Comment (NicsunXnus) [reply docs/DeveloperGuide.md:340]: Yup, I think testability will work. I thnk constraints can be used if it is hard to pinpoints exactly the nature of the requirements but here testability is mentioned so I think better go with that.

**Review by NicsunXnus — APPROVED** [review]

**Pull request merged**